common/models/user.js

```diff
@@ -607,7 +607,7 @@ module.exports = function(User) {
       type: 'email',
       from: 'noreply@example.com',
     };
-    return this.settings.verifyOptions || verifyOptions;
+    return Object.assign({}, this.settings.verifyOptions || verifyOptions);
```

Comment:
This is actually a wrong place where to fix the problem. We need to change `User.verify` so that it does not modify the provided `verifyOptions` argument.